src/index.c

```diff
@@ -513,6 +523,8 @@ static void II_SortChildren(IntersectIterator *ctx) {
 
     if (curit->mode == MODE_UNSORTED) {
       unsortedIts = array_ensure_append(unsortedIts, &curit, 1, IndexIterator *);
+    } else {
+      sortedIts = array_ensure_append(sortedIts, &curit, 1, IndexIterator *);
```

Comment:
Prefer you use malloc/free (or rm_) instead of array_. We should not end up in a situation where a member variable might use a different allocator depending on its state in the code.

Reply:
But it need to be re-sizable!!

Reply:
The array will never be larger than the total number of iterators, and we keep the size somewhere else, no?